### PR TITLE
[NGPDCP-2599 & NGPDCP-2600] Drawer component - normal and H5 variants

### PR DESCRIPTION
**Pull request by kaiceresy**
## Description

**1) Changes**

Add Drawer component for the H5 project.

IMG_1684
image

**2) Impact**

No

**3) Notes**

The H5 variant needs JS to update inline styles (`height`) for the purpose of resizing the drawer itself according to the user drag actions. So technically, this PR doesn't satisfy No. 3 of the checklist below.

## Check list

- [x] 1. Did you start and verify your changes?
- [x] 2. Did you run build to make sure that your changes can be built successfully?
- [x] 3. No !important flag, inline style in css
- [x] 4. No boilerplate codes (1)
- [x] 5. No duplicate code that exist in common functions?
- [x] 6. All of defined variables should have default value, check null and undefined?
- [x] 7. Use meaningful name for variables, no suffix 1,2,... Describes reference information for easier to understand what's inside. (2)
- [x] 9. Unsubscribed all of subscribers and even listeners when you don't need them.

```
(1)
Boilerplate codes is duplicated multiple times a bunch of the same code. This should be a common function to reuse through your code or you can use generic class / methods or design pattern to avoid the Boilerplate codes.
```

```
(2)
Follow theo coding convention
NO acronym variable name:
WRONG: el, elm
RIGHT: element
Your code can be read as a sentence
```

### CODE REVIEW

File: packages/react-compass/src/drawer/drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  })
+
+  const {child: DrawerHeaderElement, rest: OtherElementsExceptHeader} =
+    pickChild<typeof DrawerHeader>(children, DrawerHeader)
+
+  const {child: DrawerFooterElement, rest: OtherElements} = pickChild<
+    typeof DrawerFooter
+  >(OtherElementsExceptHeader, DrawerFooter)
+
+  const handleMouseDown = (e: MouseEvent<HTMLDialogElement>) => {
```
Comment (hoapn-fsoft) [inline packages/react-compass/src/drawer/drawer.tsx:113]: using `useCallback`

Comment (kaiceresy) [reply packages/react-compass/src/drawer/drawer.tsx:113]: no need to, this function will be fired only once when the user clicks on the Drawer.

File: packages/react-compass/src/drawer/drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
+'use client'
```
[the remark this answers is not in the document]
Comment (Tai-Chi-Bui) [reply packages/react-compass/src/drawer/drawer.tsx:1]: yes, we need to do it sooner or later. so you can keep it there

File: packages/react-compass/src/drawer/drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import {
+  DrawerVariantProps,
+  StyledDrawer,
+  StyledDrawerBody,
+} from './drawer.styles'
+
+interface Props extends StyledComponentProps {
+  children?: ReactNode
+  variant?: 'default' | 'h5'
+  position?: 'right' | 'bottom' | 'left' | undefined
```
[the remark this answers is not in the document]
Comment (kaiceresy) [reply packages/react-compass/src/drawer/drawer.tsx]: Adding undefined there can help users to use a state that can be `undefined` for the `position` prop without providing the Drawer's `variant`. 
Eg: Section 2. Position - drawer.stories.tsx file.
I couldn't find a proper way to declare conditional props type for react component.

File: packages/react-compass/src/drawer/drawer.tsx (hunk 1 above)
Comment (hoapn-fsoft) [reply packages/react-compass/src/drawer/drawer.tsx:133]: if `onMouseDown` props change, this function will be re-rendered.

File: packages/react-compass/src/drawer/drawer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (hoapn-fsoft) [reply packages/react-compass/src/drawer/drawer.tsx]: resolved

File: packages/react-compass/src/drawer/drawer.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hoapn-fsoft) [reply packages/react-compass/src/drawer/drawer.tsx:1]: resolved

File: packages/react-compass/src/drawer/drawer.tsx (hunk 1 above)
Comment (kaiceresy) [reply packages/react-compass/src/drawer/drawer.tsx]: updated

File: packages/react-compass/src/drawer/drawer.styles.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  insetBlock: 0,
+
+  maxHeight: '100%',
+  maxWidth: 'unset',
+
+  flexDirection: 'column',
+
+  border: 'none',
+  boxSizing: 'border-box',
+  zIndex: '50',
```
Comment (hoapn-fsoft) [inline packages/react-compass/src/drawer/drawer.styles.ts:40]: i think z-index needs to be less than 49
https://www.figma.com/file/MQVnynY3v1pI55Cm6l5qHK/CDGNotes?type=design&node-id=1-63&mode=design&t=5zW5qLdhfzUFBZ2R-0

Comment (hoapn-fsoft) [reply packages/react-compass/src/drawer/drawer.styles.ts:40]: @kaiceresy let's test with modal case, i think modal will be above drawer

**Review by hoapn-fsoft — APPROVED** [review]

**Pull request merged**